### PR TITLE
Add python interface by pybind11 and Int8 mode

### PR DESCRIPTION
**Pull request by tycallen**

### CODE REVIEW

**Comment by L1aoXingyu** [comment]
@TCHeish Please help review this PR.

**Comment by TCHeish** [comment]
> @TCHeish Please help review this PR.

Sure!

**Review by TCHeish — CHANGES_REQUESTED** [review]
1. the distill-r18 backbone part is ok.
2. the pybind11 part. 
It might require some changes, and I couldn't able to run test.py successfully.
Regarding to "batch infer now had bug" you mention in README, 
do you mean currently only single image inference works via Pybind11 ?
3. As for int8 mode, I will review soon.

Thanks!!

File: projects/FastRT/fastrt/common/common.hpp [hunk 1]
```diff
@@ -0,0 +1,324 @@
+#ifndef YOLOV5_COMMON_H_
```
Comment (TCHeish) [inline projects/FastRT/fastrt/common/common.hpp:1]: Not used in fastrt currently, please remove this file. Thanks.

File: projects/FastRT/fastrt/CMakeLists.txt [hunk 2]
```diff
@@ -2,6 +2,8 @@ project(FastRTEngine)
 
 file(GLOB_RECURSE COMMON_SRC_FILES
   ${CMAKE_CURRENT_SOURCE_DIR}/common/utils.cpp
+  ${CMAKE_CURRENT_SOURCE_DIR}/common/calibrator.cpp
+  ${CMAKE_CURRENT_SOURCE_DIR}/common/common.cpp
```
Comment (TCHeish) [inline projects/FastRT/fastrt/CMakeLists.txt:6]: Not used in fastrt currently, please remove this file. Thanks.

**Review by TCHeish — CHANGES_REQUESTED** [review]
int8 mode: 
Please remove some extra files/lines, then int8 mode is good to go!
Thanks.

File: projects/FastRT/fastrt/meta_arch/baseline.cpp [hunk 3]
```diff
@@ -19,7 +19,7 @@ namespace fastrt {
         /* Standardization */
         static const float mean[3] = {123.675, 116.28, 103.53};
         static const float std[3] = {58.395, 57.120000000000005, 57.375};
-        return addMeanStd(network, weightMap, input, "", mean, std, true); // true for div 255
+        return addMeanStd(network, weightMap, input, "", mean, std, false); // true for div 255
```
[the remark this answers is not in the document]
Comment (tycallen) [reply projects/FastRT/fastrt/meta_arch/baseline.cpp:22]: Cause the mean and std is in 0-255 range, so we don't need div 255 again. In my test only keep it false could make the model output the same like pytorch.

[the remark this answers is not in the document]
Comment (TCHeish) [reply projects/FastRT/fastrt/meta_arch/baseline.cpp:22]: I see, thanks!

[the remark this answers is not in the document]
Comment (TCHeish) [reply projects/FastRT/fastrt/meta_arch/baseline.cpp:22]: You're right, for the record. No need to div 255.
https://github.com/JDAI-CV/fast-reid/blob/9b5af4166e772f00ddc10f26fbb1f6b6546ad19a/demo/predictor.py#L51
https://github.com/JDAI-CV/fast-reid/blob/9b5af4166e772f00ddc10f26fbb1f6b6546ad19a/fastreid/modeling/meta_arch/baseline.py#L66

File: projects/FastRT/test.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
+from build.pybind_interface.ReID import ReID
```
[the remark this answers is not in the document]
Comment (tycallen) [reply projects/FastRT/test.py:1]: Seems everything is OK. Once you get that .so file, it is an importable python lib. Can you import that lib in the same level directory? `from ReID import ReID`. 
The other part I will modify as your suggestion, thank you for your careful review. After that I will find a way to provide a new dockerfile.

[the remark this answers is not in the document]
Comment (TCHeish) [reply projects/FastRT/test.py:1]: No, I can't import lib in same level directory.

[the remark this answers is not in the document]
Comment (TCHeish) [reply projects/FastRT/test.py:1]: As for INT8, I just PR a new dockerfile for it.
Last dockefile I didn't install dnn module, you may consider stack on the new one for pybind11.
Thanks.

[the remark this answers is not in the document]
Comment (TCHeish) [reply projects/FastRT/test.py:1]: I've successfully run test.py, and I will pr a new dockerfile for this(without pytorch install). 
Summery for this dockerfile: 
User can execute both c++ fastrt and test.py via pybind11 in the same container.

So maybe you can just handle the dockerfile of market_benchmark.py (which probably require stack on the env of fastreid with pybind11)
Thanks!

[the remark this answers is not in the document]
Comment (tycallen) [reply projects/FastRT/test.py]: I've finished the modify and provide a dockerfile which could run fastreid, for pybind11, users could use git command in Readme to get it.

[the remark this answers is not in the document]
Comment (TCHeish) [reply projects/FastRT/test.py]: I think it's ok now. Thanks.

**Comment by L1aoXingyu** [comment]
Good job!

**Pull request merged**